src/components/Chip/Chip.tsx

```diff
@@ -22,6 +22,29 @@ export type Props = PropsWithSpread<
      */
     appearance?: ValueOf<typeof ChipType>;
 
+    /**
+     * Whether the chip is read-only.
+     * If true, the chip will not be interactive.
+     */
+    isReadOnly?: boolean;
+    /**
+     * A badge to display on the chip.
+     */
+    badge?: ReactNode;
+    /**
+     * The name of an icon to display on the chip.
+     */
+    iconName?: string;
```

Comment:
```suggestion
    iconName?: ValueOf<typeof ICONS> | string;
```
This gives auto-completion, when using it.
Would be nice to extend the list of available strings in the `ICONS` type from `Icon.tsx` to include all the newly added icons.

Reply:
Updated the type to use the ICONS constant.

I'd propose that updating the ICONS constant be discussed separately in #1188